### PR TITLE
add documentation about Snapshots chart on Replication dashboard

### PR DESCRIPTION
**Pull request by couchand**
Add documentation for the Snapshots chart on the Replication dashboard of the Admin UI.

Still need to get a definition for "Reserved".

Happy to keep iterating on this.

Fixes: #2603

### CODE REVIEW

File: v1.1/admin-ui-replication-dashboard.md [hunk 1]
```diff
@@ -68,6 +68,19 @@ Metric | Description
 Replicas | The number of replicas.
 Quiescent | The number of replicas that haven't been accessed for a while.
 
+### Snapshots
+
+<img src="{{ 'images/admin_ui_replica_snapshots.png' | relative_url }}" alt="CockroachDB Admin UI Replica Snapshots" style="border:1px solid #eee;max-width:100%" />
+
+Usually the nodes in a Raft group stay synchronized by following along the log message by message.  However, if a node is far enough behind the log (for instance, if it was offline or is a new node getting up to speed), rather than send all the individual messages that changed the range, we can send it a snapshot of the range and it can start following along from there.
```
Comment (jseldess) [inline v1.1/admin-ui-replication-dashboard.md]: nit: `we can send it` > `the cluster can send it`

**Comment by bdarnell** [comment]
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[v1.1/admin-ui-replication-dashboard.md, line 75 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2629#-L76Stnkqd8y-V7b_Qfe:-L76Stnkqd8y-V7b_Qff:b-btc80f) ([raw file](https://github.com/cockroachdb/docs/blob/197aa57f92225ef47522c6ab9b40a9fa174f7855/v1.1/admin-ui-replication-dashboard.md#L75)):*
> ```Markdown
> CockroachDB Admin UI Replica Snapshots
> 
> Usually the nodes in a Raft group stay synchronized by following along the log message by message.  However, if a node is far enough behind the log (for instance, if it was offline or is a new node getting up to speed), rather than send all the individual messages that changed the range, we can send it a snapshot of the range and it can start following along from there.
> ```

We should talk about the preemptive case here too, since that's the most common one. 

---

*[v1.1/admin-ui-replication-dashboard.md, line 82 at r1](https://reviewable.io:443/reviews/cockroachdb/docs/2629#-L76T-B3a0OrpVLXnNJM:-L76T-B3a0OrpVLXnNJN:b7bz6k2) ([raw file](https://github.com/cockroachdb/docs/blob/197aa57f92225ef47522c6ab9b40a9fa174f7855/v1.1/admin-ui-replication-dashboard.md#L82)):*
> ```Markdown
> Normal-applied | The number of snapshots applied to nodes per second that were initiated within Raft.
> Preemptive-applied | The number of snapshots applied to nodes per second that were anticipated ahead of time (for instance, because a node was about to be added to a raft group).
> Reserved | [TO BE DETERMINED]
> ```

Before sending a snapshot, we first reserve space to ensure that we don't overload the target node. This metric snapshots that have received reservations but have not been transferred yet (i'm not sure if it goes directly from "reserved" to "applied" or if there's a time when it goes through an untracked state). 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/2629)*

File: v2.0/admin-ui-replication-dashboard.md [hunk 2]
```diff
@@ -68,6 +68,19 @@ Metric | Description
...
+
+Usually the nodes in a Raft group stay synchronized by following along the log message by message.  However, if a node is far enough behind the log (for instance, if it was offline or is a new node getting up to speed), rather than send all the individual messages that changed the range, the cluster can send it a snapshot of the range and it can start following along from there.  Commonly this is done preemptively, when the cluster can predict that a node will need to catch up, but occasionaly the Raft protocol itself will request the snapshot.
+
+Metric | Description
+-------|------------
+Generated | The number of snapshots created per second.
+Normal-applied | The number of snapshots applied to nodes per second that were initiated within Raft.
+Preemptive-applied | The number of snapshots applied to nodes per second that were anticipated ahead of time (for instance, because a node was about to be added to a raft group).
+Reserved | The number of slots reserved per second for incoming snapshots that will be send to a node.
```
Comment (couchand) [inline v2.0/admin-ui-replication-dashboard.md]: "be sent to"

**Comment by jseldess** [comment]
@couchand, do you plan to keep iterating on this, or would you rather hand it off?

**Comment by couchand** [comment]
@jseldess I think it's ready to go, pending another look from you.  Happy to make any changes requested.

File: v2.0/admin-ui-replication-dashboard.md [hunk 3]
```diff
@@ -68,6 +68,19 @@ Metric | Description
 Replicas | The number of replicas.
 Quiescent | The number of replicas that haven't been accessed for a while.
 
+### Snapshots
+
+<img src="{{ 'images/admin_ui_replica_snapshots.png' | relative_url }}" alt="CockroachDB Admin UI Replica Snapshots" style="border:1px solid #eee;max-width:100%" />
+
+Usually the nodes in a Raft group stay synchronized by following along the log message by message.  However, if a node is far enough behind the log (for instance, if it was offline or is a new node getting up to speed), rather than send all the individual messages that changed the range, the cluster can send it a snapshot of the range and it can start following along from there.  Commonly this is done preemptively, when the cluster can predict that a node will need to catch up, but occasionaly the Raft protocol itself will request the snapshot.
```
Comment (jseldess) [inline v2.0/admin-ui-replication-dashboard.md]: Let's make `Raft group` a link to the Raft portion of our architecture docs, which provide a bit more context around this concept as well as Raft logs and snapshots. 

` Usually the nodes in a [Raft group](architecture/replication-layer.html#raft)...`

File: v1.1/admin-ui-replication-dashboard.md [hunk 4]
```diff
@@ -68,6 +68,19 @@ Metric | Description
 Replicas | The number of replicas.
 Quiescent | The number of replicas that haven't been accessed for a while.
 
+### Snapshots
+
+<img src="{{ 'images/admin_ui_replica_snapshots.png' | relative_url }}" alt="CockroachDB Admin UI Replica Snapshots" style="border:1px solid #eee;max-width:100%" />
+
+Usually the nodes in a Raft group stay synchronized by following along the log message by message.  However, if a node is far enough behind the log (for instance, if it was offline or is a new node getting up to speed), rather than send all the individual messages that changed the range, the cluster can send it a snapshot of the range and it can start following along from there.  Commonly this is done preemptively, when the cluster can predict that a node will need to catch up, but occasionaly the Raft protocol itself will request the snapshot.
```
Comment (jseldess) [inline v1.1/admin-ui-replication-dashboard.md]: Let's make `Raft group` a link to the Raft portion of our architecture docs, which provide a bit more context around this concept as well as Raft logs and snapshots. 

` Usually the nodes in a [Raft group](architecture/replication-layer.html#raft)...`

File: v1.1/admin-ui-replication-dashboard.md [hunk 5]
```diff
@@ -68,6 +68,19 @@ Metric | Description
...
+
+Metric | Description
+-------|------------
+Generated | The number of snapshots created per second.
+Applied (Raft-initiated) | The number of snapshots applied to nodes per second that were initiated within Raft.
+Applied (Preemptive) | The number of snapshots applied to nodes per second that were anticipated ahead of time (for instance, because a node was about to be added to a raft group).
```
Comment (jseldess) [inline v1.1/admin-ui-replication-dashboard.md]: nit: capitalize `Raft` in `Raft group`.

File: v2.0/admin-ui-replication-dashboard.md [hunk 6]
```diff
@@ -68,6 +68,19 @@ Metric | Description
...
+
+Metric | Description
+-------|------------
+Generated | The number of snapshots created per second.
+Applied (Raft-initiated) | The number of snapshots applied to nodes per second that were initiated within Raft.
+Applied (Preemptive) | The number of snapshots applied to nodes per second that were anticipated ahead of time (for instance, because a node was about to be added to a raft group).
```
Comment (jseldess) [inline v2.0/admin-ui-replication-dashboard.md]: nit: capitalize `Raft` in `Raft group`.

File: v1.1/admin-ui-replication-dashboard.md (hunk 4 above)
Comment (jseldess) [reply v1.1/admin-ui-replication-dashboard.md]: Another nit, for the sake of consistency with other docs: Replace `for instance,` with `e.g.,`. Please make this change to the few other instances of `for instance` as well.

File: v1.1/admin-ui-replication-dashboard.md (hunk 5 above)
Comment (jseldess) [reply v1.1/admin-ui-replication-dashboard.md]: Based on the screenshot, you need to change `Applied (Preemptive)` to `Preemptive-applied`.

File: v1.1/admin-ui-replication-dashboard.md [hunk 7]
```diff
@@ -68,6 +68,19 @@ Metric | Description
...
+### Snapshots
+
+<img src="{{ 'images/admin_ui_replica_snapshots.png' | relative_url }}" alt="CockroachDB Admin UI Replica Snapshots" style="border:1px solid #eee;max-width:100%" />
+
+Usually the nodes in a Raft group stay synchronized by following along the log message by message.  However, if a node is far enough behind the log (for instance, if it was offline or is a new node getting up to speed), rather than send all the individual messages that changed the range, the cluster can send it a snapshot of the range and it can start following along from there.  Commonly this is done preemptively, when the cluster can predict that a node will need to catch up, but occasionaly the Raft protocol itself will request the snapshot.
+
+Metric | Description
+-------|------------
+Generated | The number of snapshots created per second.
+Applied (Raft-initiated) | The number of snapshots applied to nodes per second that were initiated within Raft.
```
Comment (jseldess) [inline v1.1/admin-ui-replication-dashboard.md]: Based on the screenshot, you need to change `Applied (Raf-initiated)` to `Normal-applied`.

File: v2.0/admin-ui-replication-dashboard.md [hunk 8]
```diff
@@ -68,6 +68,19 @@ Metric | Description
...
+### Snapshots
+
+<img src="{{ 'images/admin_ui_replica_snapshots.png' | relative_url }}" alt="CockroachDB Admin UI Replica Snapshots" style="border:1px solid #eee;max-width:100%" />
+
+Usually the nodes in a Raft group stay synchronized by following along the log message by message.  However, if a node is far enough behind the log (for instance, if it was offline or is a new node getting up to speed), rather than send all the individual messages that changed the range, the cluster can send it a snapshot of the range and it can start following along from there.  Commonly this is done preemptively, when the cluster can predict that a node will need to catch up, but occasionaly the Raft protocol itself will request the snapshot.
+
+Metric | Description
+-------|------------
+Generated | The number of snapshots created per second.
+Applied (Raft-initiated) | The number of snapshots applied to nodes per second that were initiated within Raft.
```
Comment (jseldess) [inline v2.0/admin-ui-replication-dashboard.md]: Based on the screenshot, you need to change `Applied (Raft-initiated)` to `Normal-applied`.

File: v2.0/admin-ui-replication-dashboard.md (hunk 6 above)
Comment (jseldess) [reply v2.0/admin-ui-replication-dashboard.md]: Based on the screenshot, you need to change `Applied (Preemptive)` to `Preemptive-applied`.

File: v2.0/admin-ui-replication-dashboard.md (hunk 3 above)
Comment (couchand) [reply v2.0/admin-ui-replication-dashboard.md]: Done.

File: v1.1/admin-ui-replication-dashboard.md (hunk 4 above)
Comment (couchand) [reply v1.1/admin-ui-replication-dashboard.md]: Done x2.

File: v1.1/admin-ui-replication-dashboard.md (hunk 5 above)
Comment (couchand) [reply v1.1/admin-ui-replication-dashboard.md]: Done.  The screenshot was out of date, I've updated.

File: v2.0/admin-ui-replication-dashboard.md (hunk 6 above)
Comment (couchand) [reply v2.0/admin-ui-replication-dashboard.md]: Done.  The screenshot was out of date, I've updated.

File: v1.1/admin-ui-replication-dashboard.md (hunk 7 above)
Comment (couchand) [reply v1.1/admin-ui-replication-dashboard.md]: The screenshot was out of date, I've updated.

File: v2.0/admin-ui-replication-dashboard.md (hunk 8 above)
Comment (couchand) [reply v2.0/admin-ui-replication-dashboard.md]: The screenshot was out of date, I've updated.

**Comment by jseldess** [comment]
Thanks, @couchand! LGTM.

**Pull request merged**